content/doc/book/blueocean/pipeline-editor.adoc

```diff
@@ -18,130 +18,115 @@ endif::[]
-The context menu (three dots on the upper right), can be used to delete the current stage.
-Clicking "Add step" will display the list of available Steps types with a search bar at the top.
-Steps can be deleted using the context <<step-configuration, context menu in the step configuration sheet>>.
-Adding a step or selecting an existing step will open the <<step-configuration, step configuration sheet>>.
+The name of the stage can be set at the top of the stage configuration pane.
+The more menu, represented by three dots to the right of the stage name, allows users to delete the currently selected stage.
+Selecting *Add step* displays the list of available step types.
+After selecting a step type, the page displays the step configuration pane.
 
 image:blueocean/editor/step-list.png[Step list filtered by 'file', role=center]
 
 == Step configuration
 
-Selecting a step from the Stage configuration sheet will open the Step Configuration sheet.
+This pane displays differently depending on the step type selected, and contains the fields or controls necessary for it.
 
 image:blueocean/editor/step-configuration.png[Step configuration for JUnit step, role=center]
 
-This sheet will differ depending on the step type,
-containing whatever fields or controls are needed.
-The name of the Step cannot be changed.
-The context menu (three dots on the upper right), can be used to delete the current step.
-Fields that contain incomplete or invalid information will display a warning symbol.
-Pipelines can have validation errors while they are being edited,
-but cannot be saved until the errors are fixed.
+After saving the name of the step, it will remain as originally entered.
+The more menu, represented by three dots to the right of the step name, allows users to delete the current step.
+Fields that contain incomplete or invalid information display a icon:exclamation[].
+Pipelines can have validation errors while editing, but saving is unavailable until there are no errors.
 
 image:blueocean/editor/step-error.png[Step configuration with error, role=center]
 
 
 == Save Pipeline dialog
 
-In order to be run, changes to a Pipeline must be saved in source control.
-The "Save Pipeline" dialog controls the saving of changes to source control.
+Changes to a Pipeline must be saved in source control before running.
+The *Save Pipeline* dialog controls the saving of changes to source control.
+
+image:blueocean/editor/save-pipeline.png[Save Pipeline dialog, role=center]
+
+There is a field to enter a description of the changes, but leaving this blank does not prevent the run.
+The dialog also provides options for saving changes to the same branch or creating a new branch to save to.
+Selecting *Save & run* saves any changes to the Pipeline as a new commit, starts a new Pipeline run based on those changes, and navigates to the <<activity#, Activity view>> for this pipeline.
+
+== Limitations
```

Comment:
It is really interesting to me that Jenkins docs adds limitations to the bottom. CBFM adds limitations to the intro, and it is in the body text, not a heading like this. No feedback, just noticing it. 😜

Reply:
updated these to below the video that appears in the first section